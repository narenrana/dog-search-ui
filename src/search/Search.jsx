import React from "react";
import PropTypes from "prop-types";
import Toolbar from "@material-ui/core/Toolbar";
import InputBase from "@material-ui/core/InputBase";
import SearchIcon from "@material-ui/icons/Search";
import useStyles from "./style";
import { search } from "../dashboard/redux";
import { useDispatch } from "react-redux";

function Search(props) {
  const dispatch = useDispatch();
  const classes = useStyles();
  const onChange = async (e) => {
    //Custome debouncing
    setTimeout(() => {
      dispatch(search(e.target.value));
    }, 1000);
  };
  return (
    <div className={classes.toolbar}>
      <Toolbar
        component="nav"
        variant="dense"
        className={classes.toolbarSecondary}
      >
        <div className={classes.search}>
          <div className={classes.searchIcon}>
            <SearchIcon />
          </div>
          <InputBase
            placeholder="Search breeds"
            classes={{
              root: classes.inputRoot,
              input: classes.inputInput
            }}
            inputProps={{ "aria-label": "Search dog breds" }}
            onChange={onChange}
          />
        </div>
      </Toolbar>
    </div>
  );
}

Search.propTypes = {
  sections: PropTypes.array,
  title: PropTypes.string
};

export default Search;
