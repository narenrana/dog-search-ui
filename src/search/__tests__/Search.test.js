/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import Search from "../Search";

import { Provider } from "react-redux";
import configureStore from "../../configure-store";

const store = configureStore();

describe("Search test", () => {
  it("Search should match snapshot", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Search />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
