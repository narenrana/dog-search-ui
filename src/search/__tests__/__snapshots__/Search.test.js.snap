// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search test Search should match snapshot 1`] = `
<div
  className="makeStyles-toolbar-1"
>
  <nav
    className="MuiToolbar-root MuiToolbar-dense makeStyles-toolbarSecondary-3 MuiToolbar-gutters"
  >
    <div
      className="makeStyles-search-8"
    >
      <div
        className="makeStyles-searchIcon-9"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <div
        className="MuiInputBase-root makeStyles-inputRoot-10"
        onClick={[Function]}
      >
        <input
          aria-label="Search dog breds"
          className="MuiInputBase-input makeStyles-inputInput-11"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search dog breds"
          type="text"
        />
      </div>
    </div>
  </nav>
</div>
`;
